File: app/src/main/java/com/example/tradeup/TradeUpApplication.java
// File: com/example/tradeup/TradeUpApplication.java

package com.example.tradeup;

import android.app.Application;
import androidx.emoji2.text.EmojiCompat;
import com.google.firebase.FirebaseApp;
import com.vanniktech.emoji.EmojiManager;
import com.vanniktech.emoji.google.GoogleEmojiProvider;
import dagger.hilt.android.HiltAndroidApp;

@HiltAndroidApp
public class TradeUpApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        EmojiCompat.init(this);
        EmojiManager.install(new GoogleEmojiProvider());
        FirebaseApp.initializeApp(this);
    }
}

File: app/src/main/java/com/example/tradeup/TradeUpApplication.kt
package com.example.tradeup // Hoặc package đúng của bạn

import android.app.Application
import android.util.Log
import dagger.hilt.android.HiltAndroidApp // Nếu bạn dùng Hilt

@HiltAndroidApp
class TradeUpApplication : Application() {

    companion object {
        private const val TAG = "TradeUpApplication"
    }

    override fun onCreate() {
        Log.e("MY_APP_DEBUG", "TradeUpApplication onCreate START") // << THÊM LOG
        super.onCreate()
        Log.d(TAG, "TradeUpApplication super.onCreate() called.")


        Log.e("MY_APP_DEBUG", "TradeUpApplication onCreate END") // << THÊM LOG
    }
}